src/core/environment.go

```diff
@@ -236,13 +238,69 @@ func (env *EnvironmentType) WriteFiles(backup ConfigApplyMarker, files []*proto.
 	}
 
 	for _, file := range files {
-		if err = writeFile(backup, file, confPath); err != nil {
+		if err = env.WriteFile(backup, file, confPath); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+// WriteFile writes the provided file content to disk. If the file.GetName() returns an absolute path, it'll be written
+// to the path. Otherwise, it'll be written to the path relative to the provided confPath.
+func (env *EnvironmentType) WriteFile(backup ConfigApplyMarker, file *proto.File, confPath string) error {
```

Comment:
can you add benchmark tests for these new functions? https://github.com/nginx/agent/blob/main/test/performance/environment_test.go#L88

Reply:
sure